### PR TITLE
Add code comment for StuckSplitTasksInterrupter race condition

### PR DESCRIPTION
**Pull request by leetcode-1533**
## Description

Add code comment for StuckSplitTasksInterrupter race condition.
More details at https://github.com/trinodb/trino/pull/13272

## Additional context and related issues
https://github.com/trinodb/trino/pull/13272

## Release notes

(x) This is not user-visible or docs only and no release notes are required.
( ) Release notes are required, please propose a release note for me.
( ) Release notes are required, with the following suggested text:

```markdown
# Section
* Fix some things. ({issue}`issuenumber`)
```

### CODE REVIEW

File: core/trino-main/src/main/java/io/trino/execution/SqlTaskManager.java
Comment (phd3) [inline core/trino-main/src/main/java/io/trino/execution/SqlTaskManager.java:712]: Can we swap order of the statement with the next one? and minor reword suggestion of  something like:

```
The detection is invoked periodically with the frequency of {@link StuckSplitTasksInterrupter#stuckSplitsDetectionInterval}.
```

File: core/trino-main/src/main/java/io/trino/execution/executor/TaskExecutor.java
Comment (phd3) [inline core/trino-main/src/main/java/io/trino/execution/executor/TaskExecutor.java:890]: nit: `TaskRunner` or `task runner`

File: core/trino-main/src/main/java/io/trino/execution/executor/TaskExecutor.java
Comment (phd3) [inline core/trino-main/src/main/java/io/trino/execution/executor/TaskExecutor.java:891]: It has a Thread object that gets assigned ...

File: core/trino-main/src/main/java/io/trino/execution/executor/TaskExecutor.java
Comment (phd3) [inline core/trino-main/src/main/java/io/trino/execution/executor/TaskExecutor.java:892]: How about we  reword this as 

```
However, when the TaskRunner moves to a different split, the thread stored here will not remain assigned to this split anymore.  
```

File: core/trino-main/src/main/java/io/trino/execution/executor/TaskExecutor.java
Comment (phd3) [inline core/trino-main/src/main/java/io/trino/execution/executor/TaskExecutor.java:492]: I think since it implements `Runnable`, the first part is kind of clear. May be just say something like below or remove

```
Keeps processing split one-by-one from the list of waiting splits.
```

File: core/trino-main/src/main/java/io/trino/execution/SqlTaskManager.java
Comment (phd3) [inline core/trino-main/src/main/java/io/trino/execution/SqlTaskManager.java:718]: thanks for adding these details!

**Comment by phd3** [comment]
cc @arhimondr

**Comment by leetcode-1533** [comment]
updated thanks!

File: core/trino-main/src/main/java/io/trino/execution/executor/TaskExecutor.java
Comment (leetcode-1533) [reply core/trino-main/src/main/java/io/trino/execution/executor/TaskExecutor.java]: removed.

**Review by phd3 — APPROVED** [review]
Made some minor language changes, will merge after CI goes through

**Comment by phd3** [comment]
hit #14814, before the fix was merged

**Pull request merged**
